backend/controllers/specializationsControllers.js

```diff
@@ -0,0 +1,149 @@
+const SpecializationsModel = require("../models/SpecializationsModel");
+const User = require("../models/UserModel");
+var mongoose = require("mongoose");
+
+module.exports.add = async (req, res) => {
+  try {
+    const { name, contributed_by, doctor_ids } = req.body;
+    if (!name) {
+      return res.status(400).json({ error: "Name is empty" });
+    }
+    const isValidContributed_by =
```

Comment:
No, This is not the functionality, As of now at the time of registration user can post a new specialization, if it is not exists in the database.

So, user_id at that point of time not exist I think

Reply:
Okay, i have commented the code for now.